### DISCUSSION TITLE
General Development Discussion

### DISCUSSION THREAD

**Discussion by LoganJFisher** [LoganJFisher/LaTeX-for-Gmail#17] [Development]
General Development Discussion

**Comment by Hakorr** [comment]
Hello! What kind of help is needed? You basically don't know JS at all or?

**Reply by LoganJFisher** [reply]
I know very little JS. I'm only really capable of minor tweaks.

Right now, TeXSend-Gmail is essentially finished (barring the discovery of any bugs, user requests, or any breaking updates to Gmail). As such, I'm looking to port its functionality over to work with Outlook.

The first priority is getting the userscript to run on https://Outlook.live.com/*, being triggered by the userscript extension's menu.

After that, implementing the button(s) and shortcuts comes next.

---

I put together the version of TeXSend-Outlook that is currently on the repository. It's based on the first beta version of TeXSend-Gmail. It works to compile LaTeX within a limited set of delimiters, and offers no further functionality. It's really just proof of concept, as TeXSend-Gmail has an overall drastically different "structure" that should be built from.

**Comment by ghesshu** [comment]
Hello, I believe I can assist with this. I'll begin by reviewing the codebase and focus on getting the userscript running on Outlook first. Once that's in place, I can move on to implementing the necessary buttons and shortcuts. If there's any other information I should know, feel free to let me know

**Reply by LoganJFisher** [reply]
I'm really glad to have you! I'll just ask that as you make PRs that you explain what feature was added so I can better test it before merging it in.

Thank you!

**Reply by LoganJFisher** [reply]
Do you need clarification on anything?

**Comment by jesusmartinez-garcia** [comment]
Hi,

I don't want to open the Issue again but I wanted to thank you for the extension. It would be great to see it working on Outlook given that most universities now make it impossible to redirect from Outlook to Gmail. I'll try the dollar sign and let you know if I find any issues.

Thanks,

Jesús

**Reply by LoganJFisher** [reply]
I'm glad you got it working. To be frank, most universities also may not be crazy about faculty running an unverified userscript (although it is open source and doesn't take long to audit as it's fairly short, all things considered), which is why I mentioned I'm looking into turning it into a browser extension (as that would at least involve verification from Google, Mozilla, etc.).

But yes, I agree that Outlook support would be a great priority. While my JS skills are decent, email sites have a lot of obfuscation and general weirdness though that I struggle with making heads or tails of. I'm still very much interested in working with someone who thinks they have what it takes to port this to working for Outlook.

And thanks. I don't expect any issues with the dollar signs, really. I think not including them was just a carry-over from when this userscript originally worked automatically, but I ultimately realized that it offers far better functionality if it's a manual toggle as it is now. As such, that should be a moot point. I'll give it a closer look soon and will likely end up adding support for dollar signs in an imminent update.

**Reply by jesusmartinez-garcia** [reply]
Actually, I think it would be fairly possible to get universities to take the script. At least in the UK, universities are mad about pleasing students (sometimes beyond reason) and would happily implement something if it can be argued that it helps students or the so called 'student experience'. It would be easy to stir (not much stirring to do, really) that a script that allows me to display math equations in my emails to them like I do in class is something we should have. If they see it works, it'll take a couple of months of loop jumping, but they'll take it.

**Reply by LoganJFisher** [reply]
Okay, so I worked out a way to implement dollar signs in a fairly satisfying way. With the update I'm about to push for the Gmail TeXSend userscript, they'll be enabled by default, but in the UI for your userscript manager, there will now be a button that says "Toggle $ & $$ Delimiters", which toggles off their ability to be treated as valid delimiters. The toggled state is remembered too (and can easily be toggled again if you want), not that you need to on every session. I could have added it as a button to the Gmail site, like the TeX button, but that frankly seemed excessive and would have felt like clutter. I think this is a good solution.

Hey, if you want to contact your university's IT department about their approval, be my guest. I'd love to be able to say that this is "Sanctioned by X University", although as I said — I would not expect them to respond so positively.

**Comment by arnaldomandel** [comment]
Here is a suggestion for the documentation, starting with a short story.
I recently noted that both in Chrome and Firefox, some emails got mangled.  That occurred regularly with NYTimes and substack, for instance.  It was quite strange, until I realized that some of the text looked like it was written as Math, clicked the TeX button and al went back to normal.
I had recently installed TeXSend in both browsers, was very happy with the result (kudos!), but it took me some time to realize this was a side effect.
Going back to the documentation, a saw the paragraph about the dollar sign (probably there were those in the mangled messages, I didn't check), and I have no qualms about your rationale there, no suggestion of change for the userscript.
Instead, I suggest that you add a warning (boldface or red) to the tone: 
_TeXSend may mangle some emails that have no business containing TeX notation; the effect can be simply reversed by clicking the TeX toggle to deactivate the script._ 
Thank you for a great piece of work.

**Reply by LoganJFisher** [reply]
Hey, thanks for the feedback. This actually brings a bug to my attention. TexSend should be disabled by default every time Gmail is loaded, requiring you to click the TeX button to manually toggle it on. That would avoid this issue. I'll look into resolving that.

**Reply by LoganJFisher** [reply]
Okay, fixed! I just had line 208 set to true, when it should have actually been false. Rolling out the update now. Thanks again for bringing this to my attention. Since the dollar signs were a recent addition and I actually had them disabled on my own system, I just never noticed this.

**Reply by arnaldomandel** [reply]
For many years I used the TeX for Gmail extension, until it was nixed; that sent me searching and eventually finding TeXSend.  Having to click the TeX button every time was a minor hassle, and I was glad to see TeX being the default.
Anyway, Murphy's Law rules, and a warning like the one I suggested cannot hurt, while keeping users aware of a potential mistake.

**Reply by LoganJFisher** [reply]
Hmm, okay. I see. I'll see about adding a toggle like the one for $ and $$ delimiters, which changes the default mode on loading Gmail. That would let you have it on by default, while maintaining the behavior that I think is generally more favorable.

I'll also see about adding a little disclaimer like you're suggesting.

**Reply by LoganJFisher** [reply]
Okay! Pushing the update now. I think this will put things in a state that leaves everyone happy. I'll look at adding a disclaimer note about this in a moment too.

**Reply by arnaldomandel** [reply]
You nailed it!  Now, it seems to recognize that actual TeX is present and activates itself, while keeping quiet otherwise.  I suppose it may make a wrong decision on some contrived cases, but who cares? Thanks for a very useful tool!

**Reply by LoganJFisher** [reply]
Not really sure what you mean about it "recognizing" (there's no "intelligence" to this), but I'm glad it's working in a way you find satisfying. :)

**Reply by arnaldomandel** [reply]
Naive I was.  But I realized one thing that is not documented: _the TeX toggle works separately on each tab_.  I assign labels to messages, and have several browser tabs, each assigned to messages with a specific label.  One of those tabs gets a lot of LaTeX messages, while others don't.  This works really well, as TeX is toggled differently in those tabs.

**Reply by LoganJFisher** [reply]
Well, that was unintentional but sounds convenient so I'll leave it as-is. Haha

**Comment by arnaldomandel** [comment]
Hi Logan, I noticed your answer on GitHub before seeing the email.
I checked a couple of mangled emails, and indeed $ was the culprit.
Are you still interested in seeing some of those?

**Reply by LoganJFisher** [reply]
I don't think it's necessary. I had sent that message before realizing that I could replicate it on my end. Thank you, though.

**Comment by Mahicopoli** [comment]
Thank you for TeXSend-Gmail. I have a question. In Gmail, if I want to see the formula for LaTeX code appreance I should toggle it. In this time the editing text is disabled and for editing I should  disable Latex. Is it possible to do it in the rea-time, as I type it I can see the output and when I want I disable it?

**Reply by LoganJFisher** [reply]
I'd say it's feasible, but would necessitate some major changes. I've frankly never encountered any LaTeX editor that even offers that, so it never even occurred to me.

It's an interesting idea, but I'll have to say it's out of scope for now unless someone decides to work it out and make a PR that introduces such functionality.

**Reply by Mahicopoli** [reply]
Thank you very much. For example in [TeX for Gmail](https://tex-for-gmail.en.softonic.com/chrome/extension) I have an istance view of LaTeX code.

**Reply by GTK-CODE** [reply]
I have already added this feature in my pr #21 for outlook, I can add it to Gmail once it's reviewed/tested.

**Reply by LoganJFisher** [reply]
Oh, fantastic. I'll get to the PR soon. It's high on my to-do list — I've just had a lot going on lately that has had to take priority. Sorry that it's taking me longer than usual to review.
